crengine/src/private/lvfreetypeface.cpp

```diff
@@ -1020,22 +1020,25 @@ void LVFreeTypeFace::setupHBFeatures()
 }
 #endif
 
-bool LVFreeTypeFace::getGlyphInfo(lUInt32 code, LVFont::glyph_info_t *glyph, lChar32 def_char, lUInt32 fallbackPassMask) {
+bool LVFreeTypeFace::getGlyphInfo(lUInt32 code, LVFont::glyph_info_t *glyph, lChar32 def_char,
+                                  lUInt32 fallbackPassMask, bool replace_missing, bool topLevel) {
     //FONT_GUARD
-    int glyph_index = getCharIndex(code, 0);
+    int glyph_index = getCharIndex(code, replace_missing, def_char);
     if (glyph_index == 0) {
         LVFont *fallback = getFallbackFont(fallbackPassMask);
-        if (NULL == fallback) {
-            // No fallback
-            glyph_index = getCharIndex(code, def_char);
-            if (glyph_index == 0)
-                return false;
-        } else {
-            // Fallback
-            lUInt32 passMask = fallbackPassMask | _fallback_mask;
-            return fallback->getGlyphInfo(code, glyph, def_char, passMask);
-        }
+        if (fallback == NULL)
+            return false;
```

Comment:
On the first call in the recursion chain, since replace_missing is false by default, we don't get a replacement char at all.